index.html

```diff
@@ -340,36 +340,43 @@ <h1>Shareholders of MTL and affiliates</h1>
             appendHoldersTableRow(fundInfo, table, accountRecord);
         }
       }
-
-      async function getDelegations(asset) {
-        const response =
-          await
-            fetch(
-              'https://api.stellar.expert/explorer/public/payments?' +
-              `asset=${asset.code}-${asset.issuer}&limit=200&order=desc`
-            );
-        const responseJson = await response.json();
-        const payments = responseJson._embedded.records;
-
-        let sources = {};
-        let targets = {};
-        for (const payment of payments) {
-          if (payment.from in sources)
-            continue; // keep only first met (last happened) delegation
-          switch (payment.memo) {
-            case 'delegate':
-              sources[payment.from] = payment;
-              if (!(targets[payment.to]))
-                targets[payment.to] = {};
-              targets[payment.to][payment.from] = payment;
-              break;
-            case 'undelegate':
-              sources[payment.from] = false;
-              break;
-          }
+      let cache = caches.open('cache');
+      
+      async function getDelegations(asset)
+      { 
+        if(cache.match(asset) !== undefined) //If this data is available in cache
+        {
+          return await cache.match(asset);
```

Comment:
you cannot return Response object from getDelegations; getDelegations must return Delegations object